Backend/app/provider-platform/dynamic-offer-driver-drainer/src/DBSync/Create.hs

```diff
@@ -1,191 +1,103 @@
+              return $ Left entryId
+            Right _ -> do
+              EL.logInfo ("QUERY INSERT SUCCESSFUL" :: Text) (" Insert successful for query :: " <> query <> " with streamData :: " <> TE.decodeUtf8 byteString)
+              return $ Right entryId
+        Nothing -> do
+          EL.logError ("No query generated for streamData: " :: Text) (TE.decodeUtf8 byteString)
+          return $ Left entryId
 
-    runCreateWithRecursion dbConf model dbObjects cmdsToErrorQueue entryIds index maxRetries ignoreDuplicates = do
-      res <- CDB.createMultiSqlWoReturning dbConf dbObjects ignoreDuplicates
-      case (res, index) of -- Ignore duplicate entry
-        (Right _, _) -> do
-          -- EL.logInfoV ("Drainer Info" :: Text) $ createDBLogEntry model "CREATE" (t2 - t1) (cpuT2 - cpuT1) dbObjects -- Logging group latencies
-          pure [Right entryIds]
-        (Left (ET.DBError (ET.SQLError (ET.MysqlError (ET.MysqlSqlError 1062 err))) _), _) -> do
-          EL.logInfo ("DUPLICATE_ENTRY" :: Text) ("Got duplicate entry for model: " <> model <> ", Error message: " <> err)
-          void $ publishDBSyncMetric $ Event.DuplicateEntryCreate model
-          -- Is retry delay needed here? :/
-          runCreateWithRecursion dbConf model dbObjects cmdsToErrorQueue entryIds index maxRetries True -- Should retry count be increased here? :/
-        (Left (ET.DBError (ET.SQLError (ET.PostgresError (ET.PostgresSqlError ("23505" :: Text) _ errMsg _ _))) _), _) -> do
-          EL.logInfo ("DUPLICATE_ENTRY" :: Text) ("Got duplicate entry for model: " <> model <> ", Error message: " <> errMsg)
-          void $ publishDBSyncMetric $ Event.DuplicateEntryCreate model
-          -- Is retry delay needed here? :/
-          runCreateWithRecursion dbConf model dbObjects cmdsToErrorQueue entryIds index maxRetries True -- Should retry count be increased here? :/
-        (Left _, y) | y < maxRetries -> do
-          void $ publishDBSyncMetric $ Event.QueryExecutionFailure "Create" model
-          EL.runIO $ delay =<< getRetryDelay
-          runCreateWithRecursion dbConf model dbObjects cmdsToErrorQueue entryIds (index + 1) maxRetries ignoreDuplicates -- Should we pass the same ignoreDuplicates or should we pass False here.
-        (Left x, _) -> do
-          void $ publishDBSyncMetric $ Event.QueryExecutionFailure "Create" model
-          EL.logError ("Create failed: " :: Text) (show cmdsToErrorQueue <> "\n Error: " <> show x :: Text)
-          pure [Left entryIds]
+-- | Generate an insert query for the atlas_driver_offer_bpp schema
+generateInsertForTable :: (Maybe Text, Maybe A.Object, Maybe A.Object) -> Text -> Maybe Text
+generateInsertForTable (mbModel, mbObject, mbMappings) tableName =
+  case (mbModel, mbObject) of
+    (Just _, Just object') -> do
+      let keys = HM.keys object'
+          newKeys = map (`replaceMappings` fromMaybe HM.empty mbMappings) keys
+          newKeys' = map (quote' . textToSnakeCaseText) newKeys
```

Comment:
I see that you are using `quietSnake` function for getting table column names. But sometimes its not working, we have exceptions here, we use `mkTableInstancesWithTModifier` for handle this exceptions in Beam instances:

`$(mkTableInstancesWithTModifier ''DriverQuoteT "driver_quote" [("requestId", "search_request_id")])`

Did you handled this in your queries?

Generally, using raw queries looks unsafe. We should discuss this possibility with @hemantmangla and others

Reply:
here replaceMappings will do that...i'm getting those mappings in strean itself
 I have made a custom instance which will be used to get all these mismatched field mappings and then i send it in streamdata